README.md

```diff
@@ -9,3 +9,38 @@ and does not require attribution for binary use.
 
 The project website is www.boost.org, where you can obtain more information and
 [download](https://www.boost.org/users/download/) the current release.
+
+## Cloning
+
+Clone the superproject repository with all submodules and checkout the `master` as the default branch:
+
+```console
+git clone --recurse-submodules https://github.com/boostorg/boost.git boost-root
+```
+
+or checkout the `develop` branch with the latest development snapshot:
+
+```console
+git clone --branch develop --recurse-submodules https://github.com/boostorg/boost.git boost-root
+```
+
+Alternatively, clone just the superproject repository, then clone a desired submodule together with its dependencies:
+
+```console
+git clone https://github.com/boostorg/boost.git boost-root
+cd boost-root
+git submodule update --init tools/boostdep
```

Comment:
one-liner `git clone --recurse-submodules="tools/boostdep" https://github.com/boostorg/boost.git boost-root`

Reply:
The aim was not to show off a clever git fu but present self-explanatory examples that even those who just know git basics can understand yhe differences between 2-3 possible or common workflows.